### DISCUSSION TITLE
是否考虑加入操作随机延时防止被ban

### DISCUSSION THREAD

**Discussion by wllsc** [NateScarlet/auto-derby#33] [Ideas]
是否需要考虑加入操作随机延时防止被ban？不知道dmm对于脚本的检查有多严格？
可以抽象出actions，在操作前加入随机delay timeout，点击操作时加入随机xy轴偏移量。

**Comment by NateScarlet** [comment]
不会加 #4 有人提了 但是你不知道它检测机制就开始搞反检测属于过早优化 有没有效果也测不出来 只是加大自己工作量而已

而且要 ban 也是 cygames 轮不到 dmm

我的[另一个 cygames 游戏辅助](https://github.com/NateScarlet/auto-dragalia) 实现的功能最后一部分都变成官方内置了

如果真 ban 了代表脚本影响太大 开源项目想反检测对面直接看我代码基本无解 大概率直接停止维护

**Comment by 1415desu** [comment]
其实从打针这个bug可以不负责任地推测一下，只有当你点击了之后才与cy服务器通信，这么说的话加入随机xy轴偏移量之类的有点掩耳盗铃了，而且感觉不容易被侦测到。

**Reply by NateScarlet** [reply]
真要反脚本最基本也是上验证码 固定点击位置检测实现麻烦反制容易 脑抽了才会这么干……

**Comment by NateScarlet** [comment]
事实上cy好像脑抽了 现在客户端会回传按钮点击位置

1.26.3 的所有点击都改成随机了
